services/soldier.service.ts

```diff
@@ -0,0 +1,30 @@
+import { z } from "zod";
+import { Soldier, SoldierQuery, Soldiers } from "../services/schemas/soldier.zschema";
+import {insertSoldier, readSoldierByID, readSoldiersByQuery} from "../repositories/soldier.repository"
+import {DocumentNotFoundError} from "./errors";
+
+async function createSoldier(doc: unknown) {
+    const demands = Soldier.parse(doc);
+    const DBResult = await insertSoldier(demands);
+    const result = Soldiers.parse(DBResult);
+    return result;
+}
+
+async function findSoldierByID(doc: unknown) {
+    const id = z.string().parse(doc);
+    const DBResult = await readSoldierByID(id);
+    if(DBResult == null){
```

Comment:
use `===` instead of `==`, but how about just `!DBResult`? Is there any value that won't work with this?

Reply:
it'll work but IMO this is more readable